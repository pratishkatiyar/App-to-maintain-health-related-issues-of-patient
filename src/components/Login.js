import React, { useState, useEffect } from 'react'
import { Redirect } from 'react-router-dom'
import Navbar from './NavBar';

const Login = () => {
    const [state, setState] = React.useState( { physicianID: "", password: "" } );
    const [loggedIn, setloggedIn] = useState(false);
    const [phydata, setphydata] = useState(null);
    
    function handleChange(evt) {
        evt.preventDefault();
        const value = evt.target.value;
        setState({
          ...state,
          [evt.target.name]: value
        });
    }

    useEffect(() => {
        getData();    
        async function getData() {
            const response = await fetch("http://localhost:3500/physicians");
            const data = await response.json();
    
            setphydata(data);
        }
      }, []);

    function submitLogin(e){
        e.preventDefault();
        let logIn=false;
        for (let item of phydata){
            if(item.physicianID===parseInt(state.physicianID) && item.password===state.password){
                setloggedIn(true);
                logIn=true;
                const now = new Date();
                const item = {
                    value: new Date().getTime().toString(),
                    expiry: now.getTime() + 86400000*2
                    // expiry: now.getTime() + 5000
                }
                localStorage.setItem("token", JSON.stringify(item));
            }
        }
        if(logIn===false) alert("Please enter correct physician login credentials!!")  
    }   	
    
    if(loggedIn===true || localStorage.getItem("token")!=null){
        return <Redirect to="/dashboard/" />
    }

    return (
        <div data-testid='login-1'>
            <Navbar/>
            <div className='dashdiv'>
                <form action='' onSubmit={submitLogin}>    
                    <label><b>Physician ID: </b></label>    
                    <input type="text" name="physicianID" placeholder="Physician ID.." value={state.physicianID} onChange={handleChange} /> 
                    <br/><br/>
                    <label><b>Password: </b></label>    
                    <input type="Password" name="password" placeholder="Password.." value={state.password} onChange={handleChange} />
                    <br/><br/> 
                    <input type="button" type="submit" value="Login" onChange={handleChange} />     
                </form>
            </div>
        </div>
    )
}

export default Login
