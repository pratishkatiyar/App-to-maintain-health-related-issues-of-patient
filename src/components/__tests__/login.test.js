import { render, screen, cleanup } from '@testing-library/react'
import { BrowserRouter, Route, Switch } from 'react-router-dom';
import Login from '../Login'
import renderer, { ReactTestRendererJSON } from 'react-test-renderer';

afterEach(() => {
    cleanup();
})

test('should render Login component', () =>{
    render(<BrowserRouter><Login/></BrowserRouter>);
    const loginElement = screen.getByTestId('login-1');
    expect(loginElement).toBeInTheDocument();
    expect(loginElement).toHaveTextContent('Physician ID:');
})

// test('renders login button correctly', () => {
//     const {getByTestId} = render(<BrowserRouter><Login/></BrowserRouter>)
//     expect(getByTestId('loginButton')).toHaveTextContent('Login')
// })

test('matches snapshot', ()=> {
    const tree = renderer.create(<BrowserRouter><Login/></BrowserRouter>).toJSON();
    expect(tree).toMatchSnapshot();
})