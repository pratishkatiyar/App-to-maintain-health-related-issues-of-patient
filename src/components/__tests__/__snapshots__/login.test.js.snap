// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  data-testid="login-1"
>
  <div
    data-testid="navbar-1"
  >
    <nav>
      <div
        className="logo"
      >
        <p>
          <a
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </p>
      </div>
      <div
        className="logo"
      >
        Patient Registration App
      </div>
      <div />
    </nav>
  </div>
  <div
    className="dashdiv"
  >
    <form
      action=""
      onSubmit={[Function]}
    >
      <label>
        <b>
          Physician ID: 
        </b>
      </label>
      <input
        name="physicianID"
        onChange={[Function]}
        placeholder="Physician ID.."
        type="text"
        value=""
      />
      <br />
      <br />
      <label>
        <b>
          Password: 
        </b>
      </label>
      <input
        name="password"
        onChange={[Function]}
        placeholder="Password.."
        type="Password"
        value=""
      />
      <br />
      <br />
      <input
        onChange={[Function]}
        type="submit"
        value="Login"
      />
    </form>
  </div>
</div>
`;
